python-frontend/src/main/java/org/sonar/python/tree/TokenImpl.java

```diff
@@ -33,23 +33,49 @@ public class TokenImpl extends PyTree implements Token {
   private com.sonar.sslr.api.Token token;
   private List<Trivia> trivia;
 
+  private Integer line;
+  private Integer column;
+  private int includedEscapeChars;
+
   public TokenImpl(com.sonar.sslr.api.Token token) {
     this.token = token;
     this.trivia = token.getTrivia().stream().map(tr -> new TriviaImpl(new TokenImpl(tr.getToken()))).collect(Collectors.toList());
   }
 
+  public TokenImpl(com.sonar.sslr.api.Token token, int line, int column, int includedEscapeChars) {
+    this(token);
+    this.line = line;
+    this.column = column;
+    this.includedEscapeChars = includedEscapeChars;
+  }
+
   @Override
   public String value() {
     return token.getValue();
   }
 
   @Override
   public int line() {
-    return token.getLine();
+    return line != null ? line : physicalLine();
   }
 
   @Override
   public int column() {
+    return column != null ? column : physicalColumn();
+  }
+
+  @Override
+  public int physicalLine() {
+    return token.getLine();
+  }
+
+  @Override
+  public int includedEscapeChars() {
```

Comment:
Here we should add a `@CheckForNull` or maybe set a default value to `0`

Reply:
The type of the variable is int, so it is 0 by default and it can't be set to `null` manually